geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java

```diff
@@ -203,43 +205,42 @@ public void channelInactive(ChannelHandlerContext ctx) {
   }
 
   private void executeCommand(ChannelHandlerContext ctx, Command command) throws Exception {
+    RedisResponse response;
     if (isAuthenticated) {
       if (command.isOfType(RedisCommandType.SHUTDOWN)) {
         this.server.shutdown();
         return;
       }
 
       if (hasTransaction() && !command.isTransactional()) {
-        executeWithTransaction(ctx, command);
+        response = executeWithTransaction(ctx, command);
       } else {
-        executeWithoutTransaction(command);
+        response = executeWithoutTransaction(command);
       }
 
       if (hasTransaction() && command.isOfType(RedisCommandType.MULTI)) {
-        writeToChannel(
-            Coder.getSimpleStringResponse(this.byteBufAllocator, RedisConstants.COMMAND_QUEUED));
-      } else {
-        // PUBLISH responses are always deferred
-        if (command.getCommandType() != RedisCommandType.PUBLISH) {
-          ByteBuf response = command.getResponse();
-          writeToChannel(response);
-        }
+        response = RedisResponse.string(RedisConstants.COMMAND_QUEUED);
       }
-    } else if (command.isOfType(RedisCommandType.QUIT)) {
-      command.execute(this);
-      ByteBuf response = command.getResponse();
-      writeToChannel(response);
-      channelInactive(ctx);
     } else if (command.isOfType(RedisCommandType.AUTH)) {
-      command.execute(this);
-      ByteBuf response = command.getResponse();
+      response = command.execute(this);
+    } else {
+      response = RedisResponse.error(RedisConstants.ERROR_NOT_AUTH);
+    }
+
+    if (response != null) {
       writeToChannel(response);
     } else {
-      ByteBuf r = Coder.getNoAuthResponse(this.byteBufAllocator, RedisConstants.ERROR_NOT_AUTH);
-      writeToChannel(r);
+      // PUBLISH responses are always deferred
+      if (command.getCommandType() != RedisCommandType.PUBLISH) {
```

Comment:
It seems like this if expression should use "!command.isOfType(RedisCommandType.PUBLISH)".

You could get this down to a single "writeToChannel" call like so:
if (response == null && !command.isOfType(PUBLISH) {
  response = command.getResponse();
}
if (response != null) {
  writeToChannel(response);
}

Reply:
I made some small updates and I think it's better - YMMV :)